### ISSUE TITLE
DotNetCoreCLIV2: Publish command fails when the output parameter and the working directory are specified

### ISSUE THREAD

**Issue by MisinformedDNA** [microsoft/azure-pipelines-tasks#13877]
## Required Information

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: DotNetCoreCLIV2

list here (V# not needed):  
https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks/DotNetCoreCLIV2

## Environment
- Server - Azure Pipelines
- Agent - Hosted

## Issue Description

When using dotnet publish, an output parameter and a working directory. the task looks for the files in the wrong place.

### Task logs

[Pipeline definition](https://github.com/MisinformedDNA/DotNetTaskPublishWithWorkingDirectory/blob/master/azure-pipelines.yml)
[Pipeline results](https://dev.azure.com/yellowcounter/GitHub/_build/results?buildId=929&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9)
[Task logs](https://github.com/microsoft/azure-pipelines-tasks/files/5521431/logs_928.zip)

## Troubleshooting
Checkout how to troubleshoot failures and collect debug logs: https://docs.microsoft.com/en-us/vsts/build-release/actions/troubleshooting

### Error logs

>   DotNetTaskPublishWithWorkingDirectory -> /home/vsts/work/1/s/DotNetTaskPublishWithWorkingDirectory/bin/Debug/netcoreapp3.1/DotNetTaskPublishWithWorkingDirectory.dll
>   DotNetTaskPublishWithWorkingDirectory -> /home/vsts/work/1/s/DotNetTaskPublishWithWorkingDirectory/drop/DotNetTaskPublishWithWorkingDirectory/
> ##[error]Error: ENOENT: no such file or directory, open 'drop/DotNetTaskPublishWithWorkingDirectory.zip'

You can see that it is writing to one directory (drop/DotNetTaskPublishWithWorkingDirectory/) and reading from another (drop/).

**Comment by tejasd1990** [comment]
@MisinformedDNA , the error is occuring not during `dotnet publish`, but during zipping the publish results after that. The input `publishTestResults` is [set to `true`](https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/DotNetCoreCLIV2/task.json#L140) by default. To zip, you will need to give the argument as `--output $(System.DefaultWorkingDirectory/drop`(giving the argument like this would also work for publish). It appears that the zip process does not understand the relative path. Hope you are able to unblock yourselves with this.

**Comment by sermetk** [comment]
Because --output parameter is trying to find a wrong directory. I used dotnet publish task without output parameter after that I used this workaround temporarily:
```
- task: PublishBuildArtifacts@1
  inputs:
    PathtoPublish: '$(Build.Repository.LocalPath)/{yourproject}bin/{buildConfiguration}/net5.0/publish.zip'
    ArtifactName: 'dr'
    publishLocation: 'Container'
```

**Comment by ddivanshu** [comment]
Closing this issue, since user is no longer blocked. Feel free to reopen/create new issue for any other problem.
